pvlib/spectrum/mismatch.py

```diff
@@ -360,31 +378,31 @@ def spectral_factor_firstsolar(precipitable_water, airmass_absolute,
     # --- Screen Input Data ---
 
     # *** Pw ***
-    # Replace Pw Values below 0.1 cm with 0.1 cm to prevent model from
-    # diverging"
+    # Replace Pw Values below min_pw with min_pw"
     pw = np.atleast_1d(precipitable_water)
     pw = pw.astype('float64')
     if np.min(pw) < min_precipitable_water:
         pw = np.maximum(pw, min_precipitable_water)
-        warn('Exceptionally low pw values replaced with '
-             f'{min_precipitable_water} cm to prevent model divergence')
+        warn('Low pw values replaced with 'f'{min_precipitable_water} cm in '
+             'the calculation of spectral mismatch.')
 
-    # Warn user about Pw data that is exceptionally high
     if np.max(pw) > max_precipitable_water:
         pw[pw > max_precipitable_water] = np.nan
-        warn('Exceptionally high pw values replaced by np.nan: '
-             'check input data.')
+        warn('High pw values replaced with np.nan in '
+             'the calculation of spectral mismatch.')
 
     # *** AMa ***
-    # Replace Extremely High AM with AM 10 to prevent model divergence
+    # Replace Extremely High AM with max_am
     # AM > 10 will only occur very close to sunset
-    if np.max(airmass_absolute) > 10:
-        airmass_absolute = np.minimum(airmass_absolute, 10)
-
-    # Warn user about AMa data that is exceptionally low
-    if np.min(airmass_absolute) < 0.58:
-        warn('Exceptionally low air mass: ' +
-             'model not intended for extra-terrestrial use')
+    if np.max(airmass_absolute) > max_airmass_absolute:
+        airmass_absolute = np.minimum(airmass_absolute, max_airmass_absolute)
+    warn('High AMa values replaced with 'f'{max_airmass_absolute} in the'
+         ' calculation of spectral mismatch.')
```

Comment:
I'm -1 to this warning.  AM=10 corresponds to zenith= ~85, and a typical annual hourly simulation is sure to have some sun positions between 85 and 90 zenith, which means this warning would always be emitted under the most common use case.  Since this only applies to situations of very marginal interest to PV production anyway, I don't think the benefit is worth the cost.

Reply:
@kandersolar the warning is not specific to AM=10 since the user can define their own maximum, although of course 10 _is_ the default.
Would you rather this warning removed entirely? How about those associated with the min AM and min/max pw cases?

Reply:
I would remove it entirely.  I won't object to keeping the other warnings, as their default thresholds correspond to conditions which are much less likely to occur than AM>10 is, and thus the warnings would not be a constant nuisance.

Reply:
@kandersolar Resolved in [6b5f1b6](https://github.com/pvlib/pvlib-python/pull/2100/commits/6b5f1b6a3141e1e34f0272461c5e992652b5c4c6).
I agree that AM=10 is not uncommon. I was concerned the absence of a warning in this case but the existence of warnings in the other cases could mislead the user into not realising values are replaced for "high" AM when they have been. I see how such warnings could be frustrating though.

Reply:
Right, this is a "lesser of two evils" situation.  

By the way, I was suggesting only that we don't emit a warning for airmass > upper_threshold, but it looks like the latest commits remove all consideration of the airmass upper limit.  I think we should retain this part of the code:

```python
    if np.max(airmass_absolute) > 10:
        airmass_absolute = np.minimum(airmass_absolute, 10)
```